common/src/main/java/tachyon/worker/DataServerMessage.java

```diff
@@ -35,7 +35,8 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  private static final int HEADER_LENGTH = 26;
+  // TODO: convert client to netty to better deal with headers.
+  private static final int HEADER_LENGTH = 38;
```

Comment:
I opted to not change this (and all the casting in this file), in favor of disturbing as little as possible. Eventually, when the client is refactored to use netty, this class will be removed.